util/encrypt.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2017-2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package util
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"time"
+
+	"github.com/marcopeereboom/sbox"
+)
+
+// EncryptionKey wraps a 32 byte encryption key and the time when it
+// was created.
+type EncryptionKey struct {
+	Key  [32]byte // Key used for encryption
+	Time int64    // Time key was created
+}
+
+// EncodeEncryptionKey encodes EncryptionKey into a JSON byte slice.
+func EncodeEncryptionKey(ek EncryptionKey) ([]byte, error) {
+	k, err := json.Marshal(ek)
+	if err != nil {
+		return nil, err
+	}
+
+	return k, nil
+}
+
+// DecodeEncryptionKey decodes a JSON byte slice into EncryptionKey.
+func DecodeEncryptionKey(payload []byte) (*EncryptionKey, error) {
+	var ek EncryptionKey
+
+	err := json.Unmarshal(payload, &ek)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ek, nil
+}
+
+// Encrypt encrypts a byte slice with the provided version using the
+// provided key.
+func Encrypt(version uint32, key [32]byte, data []byte) ([]byte, error) {
```

Comment:
Key must be a pointer. Alternatively, you could use a target instead of passing the key in.

Reply:
Made all keys pointers.